src/components/Modal/index.test.tsx

```diff
@@ -0,0 +1,52 @@
+import { fireEvent, render } from '@testing-library/react';
+
+import Modal from 'components/Modal';
+
+describe('Modal', () => {
+  describe('given show is true', () => {
+    it('displays overlay and modal', () => {
+      const Greeting = () => <div>Welcome John Doe</div>;
+
+      const { getByRole, getByTestId } = render(
+        <Modal show={true}>
+          <Greeting />
+        </Modal>
+      );
+
+      expect(getByTestId('modal-overlay')).toHaveClass('opacity-100 z-40');
+      expect(getByRole('dialog')).toHaveClass('opacity-100 z-50');
+    });
+  });
+
+  describe('given show is false', () => {
+    it('hides overlay and modal', () => {
+      const Greeting = () => <div>Welcome John Doe</div>;
+
+      const { getByRole, getByTestId } = render(
+        <Modal show={false}>
+          <Greeting />
+        </Modal>
+      );
+
+      expect(getByTestId('modal-overlay')).toHaveClass('opacity-0 -z-50');
+      expect(getByRole('dialog')).toHaveClass('opacity-0 -z-50');
+    });
+  });
```

Comment:
Should we add 2 tests:
- When `show: true`, it displays the provided content
- When `show: false`, it hides the provided content (= the text inside `<Greeting />`)

Reply:
I'm not sure, but from my understanding, the tests would still PASS, even if I remove the `{children}` part of the Modal component. So an empty modal bug would pass through the tests, no? 🤔

Reply:
@malparty I updated in this commit [c5335a7](https://github.com/hanam1ni/react-surveyor/pull/93/commits/c5335a71735fdc379475ef6084f8fa5ff4f490cb). Now the modal test will check if children are rendered by text searching and visibility of the modal by classes checking